src/main/scala/io/druid/indexer/spark/SparkBatchIndexTask.scala

```diff
@@ -407,6 +414,46 @@ object SparkBatchIndexTask
         }
       )
 
+      log.info("Initializing emitter for metrics")
+
+      val service = task.getProperties.getProperty("com.metamx.emitter.service", "druid/batch/index")
+      val host = task.getProperties.getProperty("com.metamx.emitter.host", "127.0.0.1")
+
+      val emitter = new ServiceEmitter(service, host, createEmitter(task.getProperties, lifecycle))
```

Comment:
I also preferred using the injector at first, but then got worried that things might get mixed up if the project is going to use any of Druid's internal class that uses emitter. For example, I think `S3DataSegmentPusher` uses emitter through `EmittingLogger`, although the emitter doesn't really emit anything for `S3DataSegmentPusher` and just does the logging only. Do you think it's fine to use it?